hubblestack/extmods/fdg/process.py

```diff
@@ -138,108 +138,82 @@ def _filter(seq,
         the function outputs [3, 4] - values less than or equal to 4, greater than 1,
         not equal to 2.
     '''
-    ret = seq
     for comp, value in kwargs.iteritems():
-        ret = [x for x in ret if _compare(comp, x, value)]
+        ret = [x for x in seq if _compare(comp, x, value)]
 
     return ret
 
 
-def get_index(lst, idx=0, get_last=False, append_chained=True, chained=None):
+def get_index(index=0, starting_list=None, append_chained=True, chained=None):
     '''
-    Given a list ``lst``, return the item found at ``idx``.
+    Given a list list, return the item found at ``index``.
 
-    By default, the ``lst`` will have ``.extend()`` called on it with
-    ``chained`` as the only argument.
+    By default, ``chained`` will have ``.extend()`` called on it with
+    ``starting_list`` as the only argument.
 
     The first return value (status) will be True if the return was successful, and
     False othewise. The second argument will be the requested list element.
 
-    ``append_chained`` is set to True when ``lst`` should be extended with ``chained``.
-    If set to False, chained is ignored.
+    ``append_chained`` is set to True when ``chained`` should be extended with ``starting_list``.
+    If set to False, ``starting_list`` is ignored.
 
-    ``get_last`` is used when the last element is requested; ``idx`` is overwritten.
     '''
     if append_chained:
-        if chained:
-            lst.extend(chained)
-    ret = _get_index(lst, idx, get_last)
-    status = bool(ret)
-
-    return status, ret
-
-
-def _get_index(lst,
-               idx,
-               get_last):
-    '''
-    Return the element found at index ``idx`` in the list ``lst`` or the last element if
-    ``get_last`` is set to True.
-
-    lst
-        The input list.
-
-    idx
-        The index.
-
-    get_last
-        Set to True when the last element is requested.
-    '''
-    if get_last:
-        idx = len(lst) - 1
+        if starting_list:
+            chained.extend(starting_list)
```

Comment:
Oh yeah, that was initially the behaviour but since I made the `starting_list` an optional parameter I figured it'd make more sense to use `chained `as the default list to work with and forgot to appropriately rename the parameters